### PR TITLE
Add thread states and whether or not it is doing io

### PR DESCRIPTION
**Pull request by clly**
I'm pretty sure this is doing what I want.

### CODE REVIEW

File: hhvm.py [hunk 1]
```diff
@@ -84,6 +84,45 @@ def check(self, instance):
         self.gauge("hhvm.prof-bc", resp["prof-bc"])
         self.gauge("hhvm.opt-funcs", resp["opt-funcs"])
 
+        stats_url = "{}/status.json".format(url, auth)
```
Comment (Succo) [inline hhvm.py:87]: You need 
`stats_url = "{}/status.json{}".format(url, auth)`
To properly support the auth setting.

File: hhvm.py [hunk 2]
```diff
@@ -84,6 +84,45 @@ def check(self, instance):
...
+            raise Exception("Invalid response from {}, might be a auth issue, code {}".format(stats_url, r.status_code))
+
+        resp = r.json()
+        modes = {}
+        ioduration = 0
+        io_on = 0
+        threads = resp["status"]["threads"]
+
+        for thread in threads:
+            if not thread["mode"] in modes:
```
Comment (Succo) [inline hhvm.py:103]: Using a default dict would make this cleaner https://docs.python.org/3/library/collections.html#collections.defaultdict

File: hhvm.py [hunk 3]
```diff
@@ -84,6 +84,45 @@ def check(self, instance):
...
+                if thread["io"] == 1:
+                    io_on += 1
+                    ioduration += thread["ioduration"]
+                    self.histogram("hhvm.thread.io.duration", thread["ioduration"])
+            else:
+                self.gauge("hhvm.thread.io.idle", 1)
+
+
+        for mode in modes:
+            print mode
```
Comment (Succo) [inline hhvm.py:117]: Should be removed I think.

**Review by Succo — CHANGES_REQUESTED** [review]
I've made a few comment on the python code. Overall looks great!

Do you have doc on what status.json is supposed to return? Just so I can check precisely what you are trying to mesure.

File: hhvm.py (hunk 3 above)
Comment (clly) [reply hhvm.py:117]: Yup. Needed it there for testing and forgot to remove it.

**Comment by clly** [comment]
This is what the server status looks like: https://github.com/facebook/hhvm/blob/master/hphp/doc/server.status.

My experience with it was that the request did NOT have the I/O information when the thread was not doing I/O

File: hhvm.py (hunk 1 above)
Comment (clly) [reply hhvm.py:87]: My bad. I used requests urljoin in my code but wanted it to match what you did for the pull request but messed it up.

**Pull request closed without merging**